pkg/config/config.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/sigstore/fulcio/pkg/log"
+)
+
+type FulcioConfig struct {
+	OIDCIssuers map[string]OIDCIssuer
+}
+
+type OIDCIssuer struct {
+	IssuerURL string
+	ClientID  string
+}
+
+func ParseConfig(b []byte) (FulcioConfig, error) {
+	cfg := FulcioConfig{}
+	if err := json.Unmarshal(b, &cfg); err != nil {
+		return FulcioConfig{}, err
+	}
+	return cfg, nil
+}
+
+var DefaultConfig = FulcioConfig{
+	OIDCIssuers: map[string]OIDCIssuer{
+		"https://oauth2.sigstore.dev/auth": {
+			IssuerURL: "https://oauth2.sigstore.dev/auth",
+			ClientID:  "sigstore",
+		},
+		"https://accounts.google.com": {
+			IssuerURL: "https://accounts.google.com",
+			ClientID:  "sigstore",
+		},
+	},
+}
+
+var config *FulcioConfig
+
+func Config() FulcioConfig {
```

Comment:
why you return the config thing if that is a global variable? asking to understand and to see if this is a good practice.

maybe we not make the config global  and have something like

func (c * FulcioConfig) Config()....

Reply:
ahh, right it is global in the package.

ignore me. 🤦

Reply:
Ah yeah, I don't love this approach but I like it more than stuffing this kind of thing into context variables... Not sure if there's a better way.